lib/markdown_document.rb

```diff
@@ -0,0 +1,35 @@
+class MarkdownDocument
+  YAML_FRONT_MATTER_REGEXP = /\A(---\s*\n.*?\n?)^((---|\.\.\.)\s*$\n?)/m
+
+  attr_reader :raw_content
+
+  def initialize(raw_content)
+    @raw_content = raw_content
+  end
+
+  def front_matter
+    YAML.safe_load(@raw_content.match(YAML_FRONT_MATTER_REGEXP).to_s, permitted_classes: [Date])
+  end
+
+  delegate :[], to: :front_matter
+
+  def content
+    @raw_content.gsub(YAML_FRONT_MATTER_REGEXP, "")
```

Comment:
It probably wouldn't hurt to also memoize this value. Although under it's current use case, it seems to only used once per request.